src/xnnpack/prefetch.h

```diff
@@ -6,7 +6,38 @@
 #pragma once
 
 #ifdef _MSC_VER
-  #include <intrin.h>
+  #include <intrin.h>  
+#endif
+
+#ifdef __hexagon__
+  #include <hexagon_protos.h>
+
+  // Use fetchl2 at least several hundred cycles prior to using the data.
+  
+  // height: the number of 128 bytes to fetch.
+  // width: we set 128 bytes here.
+  // stride: we set 128 bytes here.
+  XNN_INLINE static void xnn_prefetch_to_l2_linear(const void *address, uint8_t height)
+  {
+    uint16_t h0 = (((HEXAGON_Vect32)((128) & 0xff)) << 8) | ((HEXAGON_Vect32)((height) & 0xff));
```

Comment:
Isn't `128 & 0xff` just 128?

Why HEXAGON_Vect32? I don't think any casts are needed here.

In general, I think it would be better if the API was just a number of bytes. Wouldn't that make the implementation simpler too? This expression seems like it's basically just computing the number of bytes?

Reply:
1. Done - 128 & 0xff -> 128.

2. Agree with your simpler way - 
I tried to utilize HEXAGON_V32_CREATE_H and HEXAGON_V64_CREATE_H in `hexagon_types.h` - 
```
# define HEXAGON_V32_CREATE_H(h1, h0)
  ((((HEXAGON_Vect32)(h1)) << 16) | ((HEXAGON_Vect32)((h0) & 0xffff)))

#define HEXAGON_V64_CREATE_H(h3, h2, h1, h0)
  ((((HEXAGON_Vect64)(h3)) << 48LL) | (((HEXAGON_Vect64)((h2) & 0xffff)) << 32LL) |
   (((HEXAGON_Vect64)((h1) & 0xffff)) << 16LL) | ((HEXAGON_Vect64)((h0) & 0xffff)))
```
Just verified the info is same with/without casting, so I will change like this:
For linear: `uint32_t info = uint32_t info = ((128 << 16) | ((128 & 0xffff) << 8) | (height & 0xffff));`
For box: `uint64_t info = ((h3 << 48) | ((h2 & 0xffff) << 32) | ((h1 & 0xffff) << 16) | (h0 & 0xffff));`